paasta_tools/utils.py

```diff
@@ -874,7 +874,12 @@ def _run(command, env=os.environ, timeout=None, log=False, stream=False, stdin=N
             proctimer = threading.Timer(timeout, _timeout, (process,))
             proctimer.start()
         for line in iter(process.stdout.readline, ''):
-            if stream:
+            if stream and (' paasta_serviceinit ' in command and ' status ' in command):
```

Comment:
this seems pretty fragile. Can you instead move this into a function and have unit tests?